python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py

```diff
@@ -837,16 +837,27 @@ def get_backfills(
         status: Optional[BulkActionStatus] = None,
         cursor: Optional[str] = None,
         limit: Optional[int] = None,
+        filters: Optional[BulkActionsFilter] = None,
     ) -> Sequence[PartitionBackfill]:
         check.opt_inst_param(status, "status", BulkActionStatus)
-        query = db_select([BulkActionsTable.c.body])
-        if status:
-            query = query.where(BulkActionsTable.c.status == status.value)
+        query = db_select([BulkActionsTable.c.body, BulkActionsTable.c.timestamp])
+        if status or (filters and filters.status):
+            if status and filters and filters.status and status != filters.status:
+                raise DagsterInvariantViolationError(
+                    "Conflicting status filters provided to get_backfills. Choose one of status or BulkActionsFilter.status."
+                )
+            status = status or (filters.status if filters else None)
+            if status:  # should also be non-None at this point, but pyright must be appeased
+                query = query.where(BulkActionsTable.c.status == status.value)
         if cursor:
             cursor_query = db_select([BulkActionsTable.c.id]).where(
                 BulkActionsTable.c.key == cursor
             )
             query = query.where(BulkActionsTable.c.id < cursor_query)
+        if filters and filters.created_after:
+            query = query.where(BulkActionsTable.c.timestamp > filters.created_after)
```

Comment:
@prha the time-based filters for the `RunsTable` replace the timezone `filters.created_after.replace(tzinfo=None)` but the columns in the `RunsTable` are DATETIME. In the `BulkActionsTable` they are TIMESTAMP and i found that the postgres run storage tests failed when i replaced the timezone with None. The tests all pass when i leave the tzinfo as is, but i haven't been able to find a reason why online. Do you forsee any issues with not doing the timezone conversion here?

Reply:
more info -
when we add the backfill to the DB we call `datetime_from_timestamp` which will return a UTC datetime https://github.com/dagster-io/dagster/blob/b1884c3f8b83f4d64d472c98da104fd17427e2e8/python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py#L878

In the graphene layer, I'm doing the same conversion so anything coming from the UI will also get converted to UTC and filtering will be correct https://github.com/dagster-io/dagster/pull/23682/files#diff-61fefa33db2a378b1c50f360e3aa830124007f7bc8f4195005286b529b6e60cdR388

But maybe the right thing to do is to have a custom constructor on `BulkActionFilters` that takes timestamp instead of datetime and does the conversion there so that all of the datetimes on the filter are guranteed to be converted to UTC

Reply:
I think either way works...